albumentations/augmentations/transforms.py

```diff
@@ -717,7 +718,112 @@ def apply_to_keypoint(self, keypoint, crop_height=0, crop_width=0, h_start=0, w_
+        width (int): width after crop and resize.
+        scale ((float, float)): range of size of the origin size cropped
+        ratio ((float, float)): range of aspect ratio of the origin aspect ratio cropped
+        interpolation (OpenCV flag): flag that is used to specify the interpolation algorithm. Should be one of:
+            cv2.INTER_NEAREST, cv2.INTER_LINEAR, cv2.INTER_CUBIC, cv2.INTER_AREA, cv2.INTER_LANCZOS4.
+            Default: cv2.INTER_LINEAR.
+        p (float): probability of applying the transform. Default: 1.
+
+    Targets:
+        image, mask, bboxes, keypoints
+
+    Image types:
+        uint8, float32
+    """
+
+    def __init__(self, height, width, scale=(0.08, 1.0), ratio=(0.75, 1.3333333333333333),
+                 interpolation=cv2.INTER_LINEAR,
+                 always_apply=False, p=1.0):
+
+        super(RandomSizedCropTorchVision, self).__init__(min_max_height=0, height=height, width=width,
+                                                         w2h_ratio=1.0, interpolation=interpolation,
+                                                         always_apply=always_apply, p=p)
+        self.scale = scale
+        self.ratio = ratio
+
+    def get_params_dependent_on_targets(self, params):
+        img = params['image']
+        area = img.shape[0] * img.shape[1]
+
+        for attempt in range(10):
+            target_area = random.uniform(*self.scale) * area
+            log_ratio = (math.log(self.ratio[0]), math.log(self.ratio[1]))
+            aspect_ratio = math.exp(random.uniform(*log_ratio))
+
+            w = int(round(math.sqrt(target_area * aspect_ratio)))
+            h = int(round(math.sqrt(target_area / aspect_ratio)))
+
+            if w <= img.shape[1] and h <= img.shape[0]:
+                i = random.randint(0, img.shape[0] - h)
+                j = random.randint(0, img.shape[1] - w)
+                return {
+                    'crop_height': h,
+                    'crop_width': w,
+                    'h_start': i * 1.0 / (img.shape[0] - h + 1e-10),
+                    'w_start': j * 1.0 / (img.shape[1] - w + 1e-10)
+                }
+
+        # Fallback to central crop
+        in_ratio = img.shape[1] / img.shape[0]
+        if in_ratio < min(self.ratio):
+            w = img.shape[1]
+            h = w / min(self.ratio)
+        elif in_ratio > max(self.ratio):
+            h = img.shape[0]
+            w = h * max(self.ratio)
+        else:  # whole image
+            w = img.shape[1]
+            h = img.shape[0]
+        i = (img.shape[0] - h) // 2
+        j = (img.shape[1] - w) // 2
+        return {
+            'crop_height': h,
+            'crop_width': w,
+            'h_start': i * 1.0 / (img.shape[0] - h + 1e-10),
+            'w_start': j * 1.0 / (img.shape[1] - w + 1e-10)
+        }
+
+    def get_params(self):
+        return {}
+
+    @property
+    def targets_as_params(self):
+        return ['image']
+
+
+def RandomSizedCrop(*args, mode='old', **kwargs):
```

Comment:
looks like we can pass height and width for both transforms so no need to use args/kwargs for them. for other arguments it seems like we can add here all params (with same defaults) from new version and just use args, kwargs for old.

Reply:
I see what what you mean, sounds good. What about `mode='old'` and "add here all params (with same defaults) from new version", this would be a bit strange ?

Reply:
yeah it's a bit strange but doesn't break anything (but changing to mode='torchvision' breaks). at the same time it tries to push users to new api (gently).

Reply:
oh, I see what happens, agree with you - it's strange and weird :(

Reply:
so new suggestion. if user passes `min_max_height` - run old version. otherwise run new. how it sounds?

Reply:
I'm not sure how to determine whether user passes `min_max_height`. For example, in your tests:
https://github.com/albu/albumentations/blob/448761df9a008384cf914343f25e3cfb7c4d7551/tests/test_bbox.py#L189

In this case we need to check what type of input is passed. So decision based on type of input seems not very safe ?

Reply:
> yeah it's a bit strange but doesn't break anything (but changing to mode='torchvision' breaks). at the same time it tries to push users to new api (gently).

Yes, you are right, we need to keep `mode='old'` to avoid BC breaks.